django_rq/views.py

```diff
@@ -218,6 +219,29 @@ def job_detail(request, queue_index, job_id):
         job = Job.fetch(job_id, connection=queue.connection)
     except NoSuchJobError:
         raise Http404("Couldn't find job with this ID: %s" % job_id)
+    # job should have dependency but it got deleted we need to cleanup
+    try:
+        job.dependency
+    except Exception as e:
```

Comment:
Can we be more explicit as to what exception we want to catch?

Reply:
:+1: